pkg/lifecycle/ready.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2021 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package lifecycle
+
+import (
+	"sync/atomic"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	_notReady = 0
+	_ready    = 1
+)
+
+// vars
+var (
+	ErrServiceNotReady = errors.New("service is not ready to turn on/off")
```

Comment:
the error definition is confusing

Reply:
i also feel so, any good suggestion for a better/fit name?

Reply:
maybe ErrWrongState?